sdcm/tester.py

```diff
@@ -662,17 +662,23 @@ def setUp(self):  # pylint: disable=too-many-branches,too-many-statements
             else:
                 self.init_nodes(db_cluster=db_cluster)
 
-            with temp_authenticator(db_cluster.nodes[0], "org.apache.cassandra.auth.PasswordAuthenticator"):
-                # running `set_system_auth_rf()` before changing authorization/authentication protocols
-                self.set_system_auth_rf(db_cluster=db_cluster)
-                if self.params.get('use_ldap'):
+            if self.params.get('use_ldap'):
+                with temp_authenticator(db_cluster.nodes[0], "org.apache.cassandra.auth.PasswordAuthenticator"):
+                    # running `set_system_auth_rf()` before changing authorization/authentication protocols
+                    self.set_system_auth_rf(db_cluster=db_cluster)
+
                     self._setup_ldap_roles(db_cluster=db_cluster)
-                if self.params.get('ldap_server_type') == LdapServerType.MS_AD:
-                    change_default_password(node=db_cluster.nodes[0],
-                                            user=self.params.get('authenticator_user'),
-                                            password=self.params.get('authenticator_password'))
-                    self.loaders.added_password_suffix = True
-                    self.monitors.added_password_suffix = True  # FIXME: Replace with strong password generation.
+                    if self.params.get('ldap_server_type') == LdapServerType.MS_AD:
+                        change_default_password(node=db_cluster.nodes[0],
+                                                user=self.params.get('authenticator_user'),
+                                                password=self.params.get('authenticator_password'))
+                        # TODO: update proper loader and monitor in multi-tenant case.
+                        #       Now it updates only the first ones all the time.
+                        self.loaders.added_password_suffix = True
+                        # TODO: Replace with strong password generation
+                        self.monitors.added_password_suffix = True
+            else:
```

Comment:
Remove `else:`, should always be done.

Reply:
but in LDAP it was done already one, why do it again ?

Reply:
Oh, I didn't notice that. Okay, that works too.

Reply:
Israel is right. "else" is needed to avoid double work in case of LDAP.